Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -1,37 +1,355 @@
+        {
+            var highest = decimal.MinValue; 
+            var allAmounts = currentPersonData[1..]; 
+            foreach (var amountString in allAmounts)
+            {
+                var isValid = decimal.TryParse(amountString, out decimal amount); 
+                if (amount > highest)
+                    
+                {
+                    highest = amount; 
+                }
+            }
+
+            return highest; 
+        }
+
+        private static decimal CheckIfMorePeopleHaveTheSameAmount(decimal highestBalanceEver, StringBuilder personWithHighestBalance, string[] currentPersonData, decimal highestCurrentPersonBalance)
+        {
+            if (highestCurrentPersonBalance > highestBalanceEver)
+            {
+                highestBalanceEver = highestCurrentPersonBalance;
+                var currentOne = personWithHighestBalance.ToString();
+
+                if (string.IsNullOrEmpty(currentOne))
+                {
+                    personWithHighestBalance.Append(currentPersonData[0]);
+                }
+                else
+                {
+                    personWithHighestBalance.Replace(currentOne, currentPersonData[0]);
+                }
+
+
+            }
+
+            return highestBalanceEver;
         }
 
         /// <summary>
         /// Return name and loss of a person with a biggest loss (balance change negative).
         /// </summary>
         public static string FindPersonWithBiggestLoss(string[] peopleAndBalances)
         {
-            return "";
+            if (IsInValidValidInput(peopleAndBalances))
+            {
+                return InValidOutput;
+            }
+
+            var HighestLossEver = decimal.MinValue;
+            var PersonWithHighestLoss = "";
+
+            for (int i = 0; i <= peopleAndBalances.Length - 1; i++)
+            {
+                var currentPersonData = peopleAndBalances[i].Split(',');
+                var allAmounts = currentPersonData[1..];
+
+                if (allAmounts.Length <= 1)
+                {
+                    return "N/A.";
+                }
+
+                var lossForCurrentPerson  =  CalculateLossForPerson(allAmounts);
+
+                if (lossForCurrentPerson > HighestLossEver)
+                {
+                    HighestLossEver = lossForCurrentPerson;
+                    PersonWithHighestLoss = currentPersonData[0];
+                }
+            }
+
+            var answer = $"{PersonWithHighestLoss} lost the most money. -¤{HighestLossEver}.";
+
+            return answer;
+
+        }
+
+        public static decimal CalculateLossForPerson(string [] data) 
+        {
+            var lossForCurrentPerson = 0M;
+
+            for (int j = 0; j < data.Length - 1; j++)
+            {
+                var isVallid = decimal.TryParse(data[j], out decimal amount1);
+                
+                var isValid = decimal.TryParse(data[j + 1], out decimal amount2);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable 'isValid'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4937470)

Reply:
I can use that variable but normal I would return "N/A." but I cannot now and any number cannot be used because it could be a legal number given in the input.
So no idea how to solve this one. So ifyou have a hint

Reply:
According to a discussion with Josmond I can do it like this : 

```
if(!decimal.TryParse(data[j] , var out result))
{
  return;
}
``` 

and get rid of the isValid all the time.